Importing libraies and reading data csv file into a Pandas dataframe

In [1]:
import pandas as pd
import numpy as np

In [2]:
# The code was removed by Watson Studio for sharing.

In [3]:
df_tor = pd.read_csv(body)
df_tor.head()

Postcode           Borough     Neighbourhood
0      M1A      Not assigned      Not assigned
1      M2A      Not assigned      Not assigned
2      M3A        North York         Parkwoods
3      M4A        North York  Victoria Village
4      M5A  Downtown Toronto      Harbourfront

Rename columns

In [4]:
df_tor.rename(columns={'Postcode':'PostalCode', 'Neighbourhood':'Neighborhood'}, inplace=True)
df_tor.head()

PostalCode           Borough      Neighborhood
0        M1A      Not assigned      Not assigned
1        M2A      Not assigned      Not assigned
2        M3A        North York         Parkwoods
3        M4A        North York  Victoria Village
4        M5A  Downtown Toronto      Harbourfront

Remove all rows that have a Borough Not Assigned

In [5]:
df_tor['Borough'].replace('Not assigned',np.nan, inplace=True)
df_tor.dropna(subset=['Borough'], axis=0, inplace=True)
df_tor.head()

PostalCode           Borough      Neighborhood
2        M3A        North York         Parkwoods
3        M4A        North York  Victoria Village
4        M5A  Downtown Toronto      Harbourfront
5        M6A        North York  Lawrence Heights
6        M6A        North York    Lawrence Manor

Combine rows with the same Postal Code <br/>
Replace Not Assigned Neighborood values with their corresponding Borough value <br/>
Randomize dataframe rows

In [6]:
df_tor=df_tor.groupby(['PostalCode','Borough'], as_index =False).agg({'Neighborhood': ', '.join})
df_tor['Neighborhood'].replace('Not assigned',df_tor['Borough'], inplace=True)
df_tor = df_tor.sample(frac=1).reset_index(drop=True)
df_tor.head(12)

PostalCode           Borough  \
0         M9N              York   
1         M6E              York   
2         M5X  Downtown Toronto   
3         M9V         Etobicoke   
4         M1P       Scarborough   
5         M1E       Scarborough   
6         M1V       Scarborough   
7         M5C  Downtown Toronto   
8         M3K        North York   
9         M1C       Scarborough   
10        M9L        North York   
11        M2R        North York   

                                         Neighborhood  
0                                              Weston  
1                                 Caledonia-Fairbanks  
2              First Canadian Place, Underground city  
3   Albion Gardens, Beaumond Heights, Humbergate, ...  
4   Dorset Park, Scarborough Town Centre, Wexford ...  
5                   Guildwood, Morningside, West Hill  
6   Agincourt North, L'Amoreaux East, Milliken, St...  
7                                      St. James Town  
8                         CFB Toronto, Downsview East  
9              Highland Creek, Rouge Hill, Port Union  
10                                      Humber Summit  
11                                    Willowdale West

In [7]:
df_tor.shape

(103, 3)

The datframe has 103 rows or records and 3 columns or features